diff --git a/node_modules/internetarchive-sdk-js/dist/HttpClient.js b/node_modules/internetarchive-sdk-js/dist/HttpClient.js
index d9f9575..fb896be 100644
--- a/node_modules/internetarchive-sdk-js/dist/HttpClient.js
+++ b/node_modules/internetarchive-sdk-js/dist/HttpClient.js
@@ -53,11 +53,13 @@ class HttpClient {
                         }
                     }
                 }
+
                 const response = await fetch(apiUrl, {
                     headers,
                     method: endpoint.method,
                     ...(body && { body: JSON.stringify(body) }),
                     ...(data && { body: endpoint.schema.type === 'qs' ? qs.stringify(data) : data }),
+                    next: { revalidate: 3600 }
                 });
                 if (!response.ok) {
                     const message = response.status === 403 ? 'archive.org token is incorrect or you do not have access to this collection.' : endpoint?.emptyBody ? response.statusText : JSON.parse(await response.text())?.error ?? response.statusText;
diff --git a/node_modules/internetarchive-sdk-js/dist/index.cjs b/node_modules/internetarchive-sdk-js/dist/index.cjs
index ed62429..c8d44f5 100644
--- a/node_modules/internetarchive-sdk-js/dist/index.cjs
+++ b/node_modules/internetarchive-sdk-js/dist/index.cjs
@@ -149,6 +149,8 @@ class HttpClient {
                     method: endpoint.method,
                     ...(body && { body: JSON.stringify(body) }),
                     ...(data && { body: endpoint.schema.type === 'qs' ? qs.stringify(data) : data }),
+                    next: { revalidate: 3600 }
+
                 });
                 if (!response.ok) {
                     const message = response.status === 403 ? 'archive.org token is incorrect or you do not have access to this collection.' : endpoint?.emptyBody ? response.statusText : JSON.parse(await response.text())?.error ?? response.statusText;
@@ -330,33 +332,42 @@ class InternetArchive {
      */
     async getItems(items) {
         const { filters, options } = items;
-        const { fields, rows } = options || {};
+        const { fields, rows, page, sort } = options || {};
+        const limit = Number(rows) || 50;
+        const currentPage = Number(page) || 1;
+
+
         const params = {
             'q': filters?.collection && filters?.subject && filters?.creator
-                ? `collection:(${filters?.collection})&subject:("${filters?.subject}")&creator:("${filters?.creator}")`
+                ? `collection:(${filters.collection}) AND subject:("${filters.subject}") AND creator:("${filters.creator}")`
                 : filters?.collection && filters?.subject
-                    ? `collection:(${filters.collection})&subject:("${filters.subject}")`
+                    ? `collection:(${filters.collection}) AND subject:("${filters.subject}")`
                     : filters?.collection && filters?.creator
-                        ? `collection:(${filters.collection})&creator:("${filters.creator}")`
+                        ? `collection:(${filters.collection}) AND creator:("${filters.creator}")`
                         : filters?.subject && filters?.creator
-                            ? `subject:(${filters.subject})&creator:("${filters.creator}")`
+                            ? `subject:("${filters.subject}") AND creator:("${filters.creator}")`
                             : filters?.collection
                                 ? `collection:(${filters.collection})`
                                 : filters?.subject
-                                    ? `subject:(${filters.subject})`
+                                    ? `subject:("${filters.subject}")`
                                     : filters?.creator
-                                        ? `creator:(${filters.creator})`
+                                        ? `creator:("${filters.creator}")`
                                         : null,
             ...(fields && { 'fl[]': fields.replace(/ /g, '') }),
-            'rows': Number(rows) || 50,
+            'rows': limit,
+            'page': currentPage,
             'output': 'json',
-            'sort[]': 'date desc',
+            'sort[]': sort || 'date desc',
         };
+
         if (!params.q) {
-            throw new Error('collection, subject, or creator required');
+            throw new Error('At least one of collection, subject, or creator is required');
         }
+
         return await this.httpClient.makeRequest(endpoints.getItems, { params });
     }
+
+
     /**
      * Returns an Item by identifier.
      *
diff --git a/node_modules/internetarchive-sdk-js/dist/index.js b/node_modules/internetarchive-sdk-js/dist/index.js
index 17db2df..82583ac 100644
--- a/node_modules/internetarchive-sdk-js/dist/index.js
+++ b/node_modules/internetarchive-sdk-js/dist/index.js
@@ -88,31 +88,38 @@ class InternetArchive {
      */
     async getItems(items) {
         const { filters, options } = items;
-        const { fields, rows } = options || {};
+        const { fields, rows, page, sort } = options || {};
+        const limit = Number(rows) || 50;
+        const currentPage = Number(page) || 1;
+
+
         const params = {
             'q': filters?.collection && filters?.subject && filters?.creator
-                ? `collection:(${filters?.collection})&subject:("${filters?.subject}")&creator:("${filters?.creator}")`
+                ? `collection:(${filters.collection}) AND subject:("${filters.subject}") AND creator:("${filters.creator}")`
                 : filters?.collection && filters?.subject
-                    ? `collection:(${filters.collection})&subject:("${filters.subject}")`
+                    ? `collection:(${filters.collection}) AND subject:("${filters.subject}")`
                     : filters?.collection && filters?.creator
-                        ? `collection:(${filters.collection})&creator:("${filters.creator}")`
+                        ? `collection:(${filters.collection}) AND creator:("${filters.creator}")`
                         : filters?.subject && filters?.creator
-                            ? `subject:(${filters.subject})&creator:("${filters.creator}")`
+                            ? `subject:("${filters.subject}") AND creator:("${filters.creator}")`
                             : filters?.collection
                                 ? `collection:(${filters.collection})`
                                 : filters?.subject
-                                    ? `subject:(${filters.subject})`
+                                    ? `subject:("${filters.subject}")`
                                     : filters?.creator
-                                        ? `creator:(${filters.creator})`
+                                        ? `creator:("${filters.creator}")`
                                         : null,
             ...(fields && { 'fl[]': fields.replace(/ /g, '') }),
-            'rows': Number(rows) || 50,
+            'rows': limit,
+            'page': currentPage,
             'output': 'json',
-            'sort[]': 'date desc',
+            'sort[]': sort || 'date desc',
         };
+
         if (!params.q) {
-            throw new Error('collection, subject, or creator required');
+            throw new Error('At least one of collection, subject, or creator is required');
         }
+
         return await this.httpClient.makeRequest(endpoints.getItems, { params });
     }
     /**
